addons/web/models/ir_http.py

```diff
@@ -175,6 +184,24 @@ def _content_image_get_response(self, status, headers, image_base64, model='ir.a
         response.status_code = status
         return response
 
+    @api.model
+    def _content_video_get_response(self, status, headers, video_base64, download=None):
+        if status in [301, 304] or (status != 200 and download):
+            return self._response_by_status(status, headers, video_base64)
+        placeholder_content = None
+        if not video_base64:
+            # Show placeholder image if video should've 404'd
+            placeholder_content = self._placeholder()
```

Comment:
This puts a `png` binary as the response for a video `src` within a `<video>` tag.
I think you will need a single frame `placeholder.mp4` if you want to indicate a missing video through this mechanism.

Reply:
Yes I'm still testing this but it should be an mp4 :D